src/applications/vaos/services/appointment/transformers.v2.js

```diff
@@ -170,7 +170,7 @@ export function isPastAppointment(appt) {
  * @param {*} appt VAOS Service appointment object
  * @param {*} isRequest is appointment a request
  */
-export function isUpcomingAppointment(appt, isRequest) {
+export function isFutureAppointment(appt, isRequest) {
```

Comment:
What's the difference between isUpcomingAppointment ../vaos/services/appointment/index.js:366? It looks like it does what you want (with additional checks).

Reply:
@vbahinwillit - the function you're referring to takes the already transformed appointment data as the arg. ie, `appt.vaos.appointmentType` on line 367. In this case, we need to determine if the appointment is upcoming before it is transformed (in the transformer)

Reply:
So we now have 2 sources of truth for what's an upcoming appointment and it's computed differently. We have the selector, selectUpcomingAppointments and the new redux flag - appointments.confirmed[0].vaos.isUpcomingAppointment.

Reply:
Couldn't the BackLink component use the selector?

Reply:
the `selectUpcomingAppointments` selector returns a list of upcoming appts, we don't need that here.

Reply:
@vbahinwillit - the selectUpcomingAppointments selector returns a list of upcoming appts, we don't need that here.

Reply:
@vbahinwillit - I can call `selectUpcomingAppointments ` in the backLink component and try to run a check for the appt ID for the appt details page. Having this in the transformer makes it reusable in the future.